### PR TITLE
use native enums

### LINKED ISSUE #107: const enums for enums
It would be awesome if GraphQL enums were converted to `const enum` instead of union types. That way we could write `GitHubGQL.IStatusStateEnum.SUCCESS` as a value instead of `'SUCCESS'`

### PR DESCRIPTION
**Pull request by brettjurgens**
fixes #107 

WIP

### CODE REVIEW

File: packages/from-query/src/index.ts [hunk 1]
```diff
@@ -85,7 +84,7 @@ const doIt: FromQuerySignature = (schema, query, typeMap = {}, providedOptions =
   };
 
   const handleEnum: (type: GraphQLEnumType, isNonNull: boolean) => string = (type, isNonNull) => {
-    const decl: string = formatEnum(type.getValues());
+    const decl: string = type.getValues().map(({ value }) => `'${value}'`).join(' | ');
```
Comment (brettjurgens) [inline packages/from-query/src/index.ts]: need to figure out a better approach for `from-query`

**Comment by zamiang** [comment]
This appears to make type checking actually much harder and error prone. In short it is difficult to ensure that a given string matches the enum - where it is trivial to do so for types. Though I am perhaps missing how enums are supposed to be used in Typescript?

For example,
```
// a generated enum from the GraphQL schema
enum Direction {
  left = 'left',
  right = 'right',
}

const direction: Direction = 'left'; // Type 'left' is not assignable to enum Direction
const direction = 'left' as Direction; // passes type checking
const direction = 'invalid!' as Direction; // also passes type checking!
const direction = 'left' as Direction.left; // Type '"left"' is not assignable to type 'Direction.left'.
```

vs previously 
```
type Direction = 'left' | 'right';

const direction: Direction = 'left'; // passes
const direction: Direction = 'invalid!'; // doesn't work
```

**Comment by zamiang** [comment]
Apologies for the noise, on second thought, typescript enums are the coolest thing ever. I must have missed them in the TS 2.4 announcement. There may be a problem with validating output types of functions but investigating now.

**Comment by brettjurgens** [comment]
@zamiang what version of typescript are you using?

instead of using the string literal, I'd use the enum value. This should also work fine 
```ts
const direction: Direction = Direction.left; // passes
const direction: 'left' = Direction.left // passes
console.log('left' === Direction.left) // true
```

```ts
const direction = 'invalid!' as Direction; // passes
const direction = 'invalid!' as 'left' | 'right'; // but so does this!
```

```ts
const direction = 'left' as Direction.left; // passes for me in 2.7
```

**Pull request merged**